website/content/docs/api-gateway/upgrade-specific-versions.mdx

```diff
@@ -0,0 +1,159 @@
+
+**1.** Optional, double check the current version of the consul-api-gateway with the following command
+
+```
+kubectl get deployment --namespace consul consul-api-gateway-controller -o json | jq '.spec.template.spec.containers[] | select(.name="api-gateway-controller") .image'
+```
+
+If the output looks as follows
+
+```
+"hashicorp/consul-api-gateway:0.1.0"
+```
+
+ continue following this upgrade path.
+
+**2.** Run the following command to retrieve all routes defined with a cross namespace:
+
+```
+kubectl get HttpRoute -o json -A | jq -r '.items[] | {name: .metadata.name, namespace: .metadata.namespace, kind: .kind, crossNamespaceBackendReferences: ( .metadata.namespace as $parentnamespace | .spec.rules[] .backendRefs[] | select(.namespace != null and .namespace != $parentnamespace )  )}  ' && kubectl get TcpRoute -o json -A | jq -r '.items[] | {name: .metadata.name, namespace: .metadata.namespace, kind: .kind, crossNamespaceBackendReferences: ( .metadata.namespace as $parentnamespace | .spec.rules[] .backendRefs[] | select(.namespace != null and .namespace != $parentnamespace )  )}  '
+
+```
+
+Note, the above command will retrieve all HTTPRoutes and TCPRoutes. TLSRoutes and UDPRoutes are not supported in v0.1.0
+
+If you have any breaking namespaces, your output will something like as follows. If your output is empty, you have no breaking routes and you can skip to the last step.
+
+```
+{
+  "name": "example-breaking-http-route",
+  "namespace": "example-namespace",
+  "kind": "HTTPRoute",
+  "crossNamespaceBackendReferences": {
+    "group": "",
+    "kind": "Service",
+    "name": "web-backend",
+    "namespace": "gateway-namespace",
+    "port": 8080,
+    "weight": 1
+  }
+}
+{
+  "name": "example-breaking-tcp-route",
+  "namespace": "a-different-namespace",
+  "kind": "TCPRoute",
+  "crossNamespaceBackendReferences": {
+    "group": "",
+    "kind": "Service",
+    "name": "web-backend",
+    "namespace": "gateway-namespace",
+    "port": 8080,
+    "weight": 1
+  }
+}
+```
+
+**3.** Create a ReferencePolicy to allow cross namespace traffic.
+
+Using the above output as a guide, create a [ReferencePolicy](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1alpha2.ReferencePolicy) to allow cross namespace traffic. If you've already created a ReferencePolicy you can skip this step.
+
+For example, the above output would require a reference policy that looks follows.
+
+Note, the ReferencePolicy should be created in the same namespace that its allowing traffic to
+
+<CodeBlockConfig filename="referencepolicy.yaml">
+
+
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: ReferencePolicy
+metadata:
+  name: reference-policy
+  namespace: gateway-namespace
+spec:
+  from:
+    - group: gateway.networking.k8s.io
+      kind: HTTPRoute
+      namespace: example-namespace
+    - group: gateway.networking.k8s.io
+      kind: TCPRoute
+      namespace: a-different-namespace
+  to:
+    - group: ""
```

Comment:
I actually get the following error when I try remove group and specify namespace in the to field in the ReferencePolicy spec

```
error: error validating "test.yaml": error validating data: [ValidationError(ReferencePolicy.spec.to[0]): unknown field "namespace" in io.k8s.networking.gateway.v1alpha2.ReferencePolicy.spec.to, ValidationError(ReferencePolicy.spec.to[0]): missing required field "group" in io.k8s.networking.gateway.v1alpha2.ReferencePolicy.spec.to]; if you choose to ignore these errors, turn validation off with --validate=false
```

My assumption is that this is because the ReferencePolicy assumes that its directing traffic to the same namespace that the reference policy is in, but I could be wrong about that.

Reply:
Namespace doesn't get specified for `to` since it's assumed to be the namespace that the ReferencePolicy is deployed into. You can just drop `group: ""` and should be good to go

Reply:
I get 

```error: error validating "test.yaml": error validating data: ValidationError(ReferencePolicy.spec.to[0]): missing required field "group" in io.k8s.networking.gateway.v1alpha2.ReferencePolicy.spec.to; if you choose to ignore these errors, turn validation off with --validate=false```

When I drop group

Reply:
oh interesting! I don't guess I've tried installing with group deleted. This has implications for https://github.com/hashicorp/tutorials/pull/260 then

Reply:
Just saw this, that's silly. Think we should raise this upstream. If there's default behavior for an empty string (i.e. `core`), this should just be an optional field in the spec.